Tools/CISupport/ews-build/steps.py

```diff
@@ -1361,7 +1370,17 @@ def validate_github(self, pr_number):
             self.skip_build("PR {} does not have a merge queue label".format(pr_number))
             return False
 
-        if -1 in (obsolete, pr_closed, blocked, merge_queue):
+        draft = self._is_pr_draft(pr_json) if self.verifyNoDraft else 0
+        if draft == 1:
+            reason = "PR {} has been marked as a draft".format(pr_number)
+            self._addToLog('stdio', reason)
+            self.finished(FAILURE)
```

Comment:
I did deliberately make this `FAILED` instead of `SKIPPED`.

My reason for this is that both `merge-queue` and draft PRs require explicit action by the user, and can be incorrectly combined. Our existing checks don't really have this same property, they're all primarily designed to operate on PRs which are slightly out of date.

Reply:
Then maybe it should be FAILED only for Merge Queue, and SKIPPED for all other queues.

If someone creates a draft PR, it would be weird for the user to see all the EWS builds as failing.

Reply:
Agreed on the weirdness of that behavior, that's why I set `verifyNoDraft=False` by false, so we are only running this check on `Merge-Queue`

Reply:
ok.
Can make this a function, e.g.: fail_build(), just like skip_build()